nncf/experimental/torch/sparsity/movement/structured_mask_handler.py

```diff
@@ -381,24 +401,35 @@ def _create_structured_mask_context_groups(
             for consumer in group.consumers:
                 if consumer.pruning_dimension is not None:
                     extended_producers.add(ProducerInfo(consumer.node_id, consumer.pruning_dimension))
-
+            is_group_matched = True
+            prefix_warning = 'Automatically found structured pruning group does not match the given unstructured '\
+                f'pruning structures: \n{group}\n.'
             for producer_info in extended_producers:
                 nncf_node = nncf_graph.get_node_by_id(producer_info.node_id)
                 module = nncf_network.nncf.get_containing_module(nncf_node.node_name)
                 if module in module_vs_sparse_module_info_map:
                     # 0 dimension corresponds to row (output channels), 1st dimension - to column (input channels)
                     prune_by_row = not bool(producer_info.pruning_dimension)
                     minfo = module_vs_sparse_module_info_map[module]
+                    sparsifier: MovementSparsifier = minfo.operand
+                    if sparsifier.sparse_structure == SparseStructure.PER_DIM:
+                        sparse_axis = sparsifier.sparse_cfg.sparse_axis
```

Comment:
Filter out groups that have structured pruning per dimension that doesn't match the given dimension for unstructured pruning